src/http_proxy_io.c

```diff
@@ -45,6 +48,7 @@ typedef struct HTTP_PROXY_IO_INSTANCE_TAG
     XIO_HANDLE underlying_io;
     unsigned char* receive_buffer;
     size_t receive_buffer_size;
+    bool use_tls_io;
```

Comment:
need to also add the copy code to http_proxy_io_clone_option()

Reply:
I'm doing it as `if ((strcmp(name, OPTION_UNDERLYING_IO_OPTIONS) == 0) || (strcmp(name, OPTION_USE_TLS_HTTP_PROXY) == 0))`

not sure if this is the correct/preferred way